aten/src/ATen/native/ReflectionPad.cpp

```diff
@@ -264,30 +264,28 @@ inline void reflection_pad1d_out_loop(
   });
 }
 
-void reflection_pad1d_out_template(
-    const Tensor& output, const Tensor& input_, IntArrayRef padding) {
-  int64_t dim_plane = 0;
-  int64_t dim_w = 1;
-  int64_t nbatch = 1;
+namespace {
+void reflection_pad1d_out_template_helper(const Tensor& input_, IntArrayRef padding,
+  int64_t& nbatch, int64_t& pad_l, int64_t& pad_r, int64_t& nplane, int64_t& input_w,
+  int64_t& output_w, Tensor& input, const Tensor& output) {
   // allow dim=0 only in the batch dimension.
   TORCH_CHECK(
       (input_.ndimension() == 2 && input_.size(1) != 0) ||
       (input_.ndimension() == 3 && input_.size(1) != 0 && input_.size(2) != 0),
       "2D or 3D (batch mode) tensor expected for input, but got: ", input_);
-
+  int64_t dim_plane = 0;
+  int64_t dim_w = 1;
+  nbatch = 1;
   if (input_.ndimension() == 3) {
     nbatch = input_.size(0);
     dim_w++;
     dim_plane++;
   }
-
-  /* sizes */
-  auto pad_l = padding[0];
-  auto pad_r = padding[1];
-
-  int64_t nplane = input_.size(dim_plane);
-  int64_t input_w = input_.size(dim_w);
-  int64_t output_w  = input_w + pad_l + pad_r;
+  pad_l = padding[0];
+  pad_r = padding[1];
+  nplane = input_.size(dim_plane);
+  input_w = input_.size(dim_w);
+  output_w  = input_w + pad_l + pad_r;
```

Comment:
actually I don't feel it's a good practice to pass in reference and modify them in the function, can we just copy around this code or have a function that returns these things?

Reply:
can structured kernel help here as well? cc @ezyang

Reply:
I have no idea what this code is trying to do, but yes, structured kernels let you precompute quantities like this so they are passed in directly to the function.